pwndbg/chain.py

```diff
@@ -95,20 +107,12 @@ def format(value, limit=LIMIT, code=True, offset=0, hard_stop=None, hard_end=0):
     # Otherwise, the last element in the chain is the non-pointer value.
     # We want to enhance the last pointer value. If an offset was used
     # chain failed at that offset, so display that offset.
-    elif len(chain) < limit:
+    elif len(chain) < limit + 1:
```

Comment:
Since we are no longer doing that nasty `limit = int(limit)+1` hack, you would expect that all addresses along the chain were pointers, then `len(chain) == limit + 1`.

If the chain failed to dereference any entry, then `len(chain) <= limit`, in other words, `len(chain) < limit + 1`.